src/vs/code/electron-main/app.ts

```diff
@@ -1471,4 +1488,47 @@ export class CodeApplication extends Disposable {
 		// Refs: https://github.com/microsoft/vscode/issues/234064
 		validatedevDeviceId(this.stateService, this.logService);
 	}
+
+	private async moveToApplicationsFolder(): Promise<void> {
+		const { response } = await dialog.showMessageBox(massageMessageBoxOptions({
+			type: 'info',
+			buttons: [
+				localize({ key: 'move', comment: ['&& denotes a mnemonic'] }, "&&Move to Applications"),
+				localize({ key: 'doNotMove', comment: ['&& denotes a mnemonic'] }, "&&Do not Move")
+			],
+			message: localize('moveToApplicationsFolderWarning', "{0} works best when run from the Applications folder", this.productService.nameLong),
```

Comment:
Perhaps it'd be nice to say something about the reason, e.g. "Auto-updating {0} works best..." or "{0} will not be able to auto-update unless you move it to the Applications folder"